src/Microsoft.AspNet.OData/Query/PlainTextODataQueryOptionsParser.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System.IO;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.OData;
+using Microsoft.AspNet.OData.Common;
+
+namespace Microsoft.AspNet.OData.Query
+{
+    [SuppressMessage("Microsoft.Naming", "CA1702:CompoundWordsShouldBeCasedCorrectly", Justification="Spelling correct in this context")]
+    public partial class PlainTextODataQueryOptionsParser
+    {
+        /// <inheritdoc/>
+        public string Parse(Stream requestStream)
```

Comment:
Out of curiosity, why isn't this method async?

Reply:
@habbes I also considered making it async. However, based on how this entire feature is used, I didn't find any tangible benefit that would be accrued from that. We receive a request and the determination how we should proceed to process the request needs to happen pronto. Note that the execution starts from `Match` method that is also not async. I'd be open to reconsider the approach though if making it async is the better choice

Reply:
@habbes I eventually with async processing to address an issue where a particular test was failing intermittently